`prometheus.operator.*`: Allow Customization of Default `scrape_interval` and `scrape_timeout` (#5139)
grafana/agent: docs/sources/shared/flow/reference/components/prom-operator-scrape.md

---
aliases:
- /docs/agent/shared/flow/reference/components/prom-operator-scrape/
canonical: https://grafana.com/docs/agent/latest/shared/flow/reference/components/prom-operator-scrape/
headless: true
---

Name | Type | Description | Default | Required
---- | ---- | ----------- | ------- | --------
`default_scrape_interval` | `duration` | How frequently to scrape targets. Used as default if no value is provided by the target resource. | `1m` | no

> `default_scrape_timeout` | `duration` | How long until a scrape request times out. Used as default if no value is provided by the target resource.| `10s` | no

Reviewer: What do you mean by `Used as default`? Are the arguments used as default or are the 1m and 10s values used as default if you don't specify a value for `default_scrape_interval` and default_scrape_timeout`?

Reviewer 2: Those values will be used if a given Service Monitor does not specify the interval. Currently a single hard-coded default is used in that case.

Reviewer: ```suggestion
`default_scrape_interval` | `duration` | The default interval between scraping targets. Used as the default if the target resource does not provide a scrape interval. | `1m` | no
`default_scrape_timeout` | `duration` | The default timeout for scrape requests. Used as the default if the target resource does not provide a scrape timeout. | `10s` | no
```

Does this suggestion clarify this in the description?

Author: This sounds a lot better. I updated the descriptions in 28550c2792f18b6c13f12758518f58d17aefce8d and rebased everything to resolve the merge conflicts
